service/models.py

```diff
@@ -84,19 +83,21 @@ def deserialize(self, data):
         """
         try:
             self.name = data["name"]
+            self.available = data["available"]
             self.price = data["price"]
             self.stock = data["stock"]
             self.size = data["size"]
             self.color = data["color"]
             self.category = data["category"]
-            self.gender = data["gender"]
             self.description = data["description"]
-            self.date_added = data["date_added"]
+
         except KeyError as error:
-            raise DataValidationError("Invalid product: missing " + error.args[0])
-        except TypeError as error:
             raise DataValidationError(
-                "Invalid product: body of request contained" "bad or no data"
+                "Invalid Product: missing " + error.args[0]
+            )
+        except TypeError:
```

Comment:
@Kjabraham the removed line above (line 97) has 'except TypeError as error:' while the added line just has 'except TypeError:' - what's the difference and why was 'as error' included in original code?

Reply:
the 'as error' sets the message the system throws to the variable 'error' however we then do not use that 'error' variable so it is unnecessary to declare. 

Hint: these are all changes coming from pylint running and trying to get us to 9/10

Reply:
Ok sounds good thanks. I'll ask you later what pylint is.